src/gc.rs

```diff
@@ -0,0 +1,92 @@
+use crate::Sodg;
+use anyhow::{Context, Result};
+use std::collections::VecDeque;
+
+impl Sodg {
+    /// Attempts to collect the vertex `v`.
+    ///
+    /// If there are no edges leading to `v`, then it is deleted and all its children are collected.
+    /// Otherwise, nothing happens.
+    ///
+    /// ```
+    /// use sodg::{Hex, Sodg};
+    /// let mut g = Sodg::empty();
+    /// g.add(1).unwrap();
+    /// g.put(1, Hex::from(0)).unwrap();
+    /// g.add(2).unwrap();
+    /// g.put(2, Hex::from(0)).unwrap();
+    /// g.bind(1, 2, "x").unwrap();
+    /// g.data(2).unwrap(); // Try to collect 2
+    /// assert!(g.data(2).is_ok());
+    /// g.data(1).unwrap(); // Successfully collect 1
+    /// assert!(g.data(1).is_err());
+    /// ```
+    pub(crate) fn collect(&mut self, v: u32) -> Result<()> {
+        let mut queue = VecDeque::new();
+        queue.push_back(v);
+        while !queue.is_empty() {
+            let collected_vertex_id = queue
```

Comment:
@UARTman the naming of variables makes it hard to read the code. In general, try to NOT to use compound names of variables: https://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html Call the start vertex as `start`, `collected_vertex_id` would be `v`, and `collected_vertex` would be `vtx` (this is the convention in this repo)